### ISSUE TITLE
Val accuracy not changing

### ISSUE THREAD

**Issue by allenye66** [keras-team/keras#16327]
I am trying to train a LSTM to binary classify stock market data.

My dataset contains 543 rows of data, with each row having 150 columns. The data has been standardized.

<img width="754" alt="WCWbl" src="https://user-images.githubusercontent.com/46653284/160743771-79e98c88-02b6-4078-a375-3adbff3c0d55.png">

I have ~600 samples, each has 300 time steps and each time step has just 1 feature (the price). My reshaped data is (500, 300, 1) for training, and (100, 300, 1) for testing.

My labels are either 1 or 0.
<img width="643" alt="DkB34" src="https://user-images.githubusercontent.com/46653284/160743956-624b52ae-1c51-4ac7-9682-66fd0f7ca743.png">

However, when training my model, my val accuracy never changes no matter what I try. I believe it might be the data dimensions I am passing in. Does anyone know how to improve my model?

```
X_train, X_test, y_train, y_test = train_test_split(df, labels, random_state = 0, test_size = 0.2, stratify = labels)
#train and test shape is (434, 300) and (109, 300)
X_train = tf.reshape(X_train, X_train.shape + (1,))
X_test = tf.reshape(X_test, X_test.shape + (1,))
#train and test shape is (434, 300, 1) and (300, 150, 1)


model = Sequential()
model.add(LSTM(64, dropout=0, recurrent_dropout=0.2,go_backwards= True, input_shape=(300, 1)))
model.add(Dropout(0.5))
model.add(Dense(1,activation='sigmoid'))
model.compile(loss = "binary_crossentropy", optimizer = 'adam', metrics=['accuracy'])
model.summary()

```
<img width="610" alt="Ttydr" src="https://user-images.githubusercontent.com/46653284/160744044-ff6b49a9-6d7a-4ec1-a075-c4740c7b89ed.png">

```
history = model.fit(X_train,y_train, epochs = 10,batch_size=1,validation_data = (X_test, y_test), shuffle = True)


```
<img width="1083" alt="oiBrO" src="https://user-images.githubusercontent.com/46653284/160744078-f7e4e1aa-04bf-4095-9375-57a388a096d4.png">

I've tried different models, changing the data, but the val accuracy can't seem to change.

**Comment by tilakrayal** [comment]
@allenye66 ,
The size of the training dataset you are using is less. While the amount of the trainable parameters is around 17K. This might possibly gone for overfitting.

Please try with below
1.Generate or find more data,
2.Reduce the complexity of the model,
3.Use pre-trained embedding,
4.Reduce the number of the LSTM nodes.
Also please take a look at this Keras [optimizers](https://keras.io/api/optimizers/) from the documentation.Thanks!

**Comment by ernestlwe** [comment]
hi Allen, wondering if u managed to solve this and how pls?
